Phoenix/Core/Include/Core/Voxels/Blocks.hpp

```diff
@@ -54,21 +56,22 @@ namespace phoenix
 		};
 
 		/// @brief Stores universal block definitions
-		class BlockRegistry
+		class BlockRegistry : public qz::utils::Singleton<BlockRegistry>
 		{
 		private:
 			std::vector<RegisteredBlock> Blocks;
 			int                          i;
```

Comment:
What is `i` used for? A name like `i` sounds like a counter variable so probably shouldn't be a member.